### PR TITLE
Differentiate between interfaces and structs in outline

### PR DESCRIPTION
**Pull request by karthikraobr**
This PR fixes the code outline from erroneously displaying structs as interfaces.

Before 
screenshot 2018-11-11 at 00 29 59

After
screenshot 2018-11-11 at 00 31 02

### CODE REVIEW

**Comment by karthikraobr** [comment]
@ramya-rao-a No idea why the test failed :(

File: src/goOutline.ts [hunk 1]
```diff
@@ -156,10 +156,13 @@ function convertToCodeSymbols(
 			let start = byteOffsetToDocumentOffset(decl.start - 1);
 			let end = byteOffsetToDocumentOffset(decl.end - 1);
 			range = new vscode.Range(document.positionAt(start), document.positionAt(end));
+			if (decl.type === 'type') {
+				let line = document.lineAt(document.positionAt(start));
+				let regex = new RegExp('\\bstruct\\b');
```
Comment (ramya-rao-a) [inline src/goOutline.ts]: Can you make this regex stricter? Something like `RegExp('^\s*type\s+' + label + '\s+struct\s+ {')` ?

Also, update the test for the outline feature.

Comment (karthikraobr) [reply src/goOutline.ts]: Thanks. I guess you are referring to an older version of my code. But, as you pointed out even the newer exp is not very strict. Will fix it and fix the tests.

Comment (karthikraobr) [reply src/goOutline.ts]: For some reason, regex.test is always returning false if I use the above regex (IMO it is right). I even tried  ```let regex = new RegExp(`^\s*type\s${label}\s+struct\s*{`);```.  Still returns false.

**Comment by ramya-rao-a** [comment]
@karthikraobr I think I am missing something here...

Below is the screenshot from using the latest Go extension

image

The interface and struct are appearing separately for me without the changes in this PR.

What am I missing?

**Comment by karthikraobr** [comment]
@ramya-rao-a I am on the latest version of vscode and the plugin as well. But doesn't work for me. Moreover, a struct is being displayed as a class which is wrong. Finally, there is no [code](https://github.com/Microsoft/vscode-go/blob/e05d03579d33b882636e11e27bb4094b3e294cee/src/goOutline.ts#L124) to detect classes right? Are you using a beta?

**Comment by karthikraobr** [comment]
@ramya-rao-a Thanks a ton for fixing the regex. I have added tests for struct outline. Let me know if I need to add more tests.

**Comment by karthikraobr** [comment]
This PR also fixes breadcrumbs for go files.

**Comment by ramya-rao-a** [comment]
Thanks @karthikraobr!

**Pull request merged**